src/player.js

```diff
@@ -647,6 +647,26 @@ export default class Player extends FakeEventTarget {
     }
   }
 
+  get stats(): PKStatsObject {
+    const videoElement = this.getVideoElement();
+    let statsObject: PKStatsObject = {
+      targetBuffer: NaN,
+      availableBuffer: NaN
+    };
+    if (this._engine) {
+      statsObject.targetBuffer = this._engine.targetBuffer;
+      statsObject.availableBuffer = this._engine.availableBuffer;
+      // consideration of the end of the playback in the target buffer calc
+      if (!this.isLive()) {
```

Comment:
I think you can simplify this to 
```javascript 
if (!this.isLive() && videoElement) {
  statsObject.targetBuffer = Math.min(statsObject.targetBuffer, videoElement.duration - videoElement.currentTime);
}
```
You can also move the `const videoElement = this.getVideoElement();`  inside the of as you only use it there 

Also, did you add the new engine members to flash and YouTube engines?

Reply:
Also, shouldn’t this be `if ((!this.isLive() || this.isDvr()) && vidoeElement)`

Reply:
The dvr handling is being done in HLS and Dash as this depends on the player implementation

Reply:
> I think you can simplify this to
> 
> ```js
> if (!this.isLive() && videoElement) {
>   statsObject.targetBuffer = Math.min(statsObject.targetBuffer, videoElement.duration - videoElement.currentTime);
> }
> ```
> 
> You can also move the `const videoElement = this.getVideoElement();` inside the of as you only use it there
> 
> Also, did you add the new engine members to flash and YouTube engines?

Nope, needs to be done...